soem/ethercatmain.c

```diff
@@ -347,7 +347,7 @@ uint8 ecx_siigetbyte(ecx_contextt *context, uint16 slave, uint16 address)
    if (address < EC_MAXEEPBUF)
    {
       mapw = address >> 5;
-      mapb = address - (mapw << 5);
+      mapb = address - (uint16)(mapw << 5);
```

Comment:
make similar as in line 378 : (uint16)((eadr << 1) - (mapw << 5));

Reply:
fixed